options/options.go

```diff
@@ -107,7 +114,7 @@ func InterpretOptions(linked linker.Result, handler *reporter.Handler, opts ...I
 // Any options that are uninterpretable (due to such errors) will remain in the
 // "uninterpreted_option" fields.
 func InterpretOptionsLenient(linked linker.Result, opts ...InterpreterOption) (sourceinfo.OptionIndex, error) {
-	return interpretOptions(true, linked, linker.ResolverFromFile(linked), reporter.NewHandler(nil), opts)
+	return interpretOptions(true, linked, linker.ResolverFromFile(linked), nil, opts)
```

Comment:
No need to allocate a reporter like this in these lenient mode functions: we instead use a special handler implementation that behaves like `reporter.NewHandler(nil)` except in the places that need more lenience.